### ISSUE TITLE
Characters : and @ are not escaped in usernames/passwords

### ISSUE THREAD

**Issue by traut** [gruns/furl#56]
`@` and `:` are not escaped in result URL when they are present in username/password:

```python
> from furl import furl
> f = furl('http://some-domain.com')
> f.username = 'username@withcharacter'
> f.password = 'complex@password:characters'
> print(f.url)

http://username@withcharacter:complex@password:characters@some-domain.com
```

According to [RFC 3986](http://www.faqs.org/rfcs/rfc3986.html), section 3.2.1, it needs to be percent encoded:
```
  userinfo    = *( unreserved / pct-encoded / sub-delims / ":" )
```
So it should look like:
```
http://username%40withcharacter:complex%40password%3Acharacters@some-domain.com
```

**Comment by gruns** [comment]
Great find. I'll fix this shortly.

**Comment by moloch--** [comment]
There's a great chapter in [The Tangled Web](http://lcamtuf.coredump.cx/tangled/) on this, browsers handle it differently. For example: `http://foo.com\@bar.com/` in Chrome/WebKit you'll end up at `foo.com` in Firefox you'll end up at `bar.com`.

**Comment by gruns** [comment]
Fixed in trunk. This fix will be included in the next version, v0.4.9.

Thank you for bringing this issue to my attention, Sergey. And thank you for your input, @moloch--.

**Comment by traut** [comment]
thanks!

**Comment by gruns** [comment]
Fix included in furl v0.4.9. Upgrade with

```
pip install furl --upgrade
```

Thank you again Sergey and @moloch--.
